godtools/App/Share/Data/FavoritedResourcesRepository/Cache/RealmFavoritedResourcesCache.swift

```diff
@@ -117,11 +143,109 @@ class RealmFavoritedResourcesCache {
+            
+            let resourcesToMoveUp = realm.objects(RealmFavoritedResource.self).where({ $0.position >= positionToDelete })
+            for resource in resourcesToMoveUp {
+                resource.position -= 1
+            }
+            
+            return Array(resourcesToMoveUp)
+            
+        } mapInBackgroundClosure: { objects in
+            return []
+        }
+        .flatMap { _ in
+            return self.realmDatabase.deleteObjectsInBackgroundPublisher(
+                type: RealmFavoritedResource.self,
+                primaryKeyPath: #keyPath(RealmFavoritedResource.resourceId),
+                primaryKeys: [id]
+            )
+        }
         .eraseToAnyPublisher()
     }
+    
+    func reorderFavoritedResourcePublisher(id: String, originalPosition: Int, newPosition: Int) -> AnyPublisher<[FavoritedResourceDataModel], Error> {
+        
+        return realmDatabase.writeObjectsPublisher { realm in
+            var resourcesToUpdate: [RealmFavoritedResource] = []
+            
+            if newPosition - originalPosition > 0 {
+                let resourcesToMoveUp = realm.objects(RealmFavoritedResource.self).where({ $0.position >= originalPosition && $0.position <= newPosition })
+                for resource in resourcesToMoveUp {
+                    resource.position -= 1
+                }
+                
+                resourcesToUpdate += resourcesToMoveUp
+            } else {
+                let resourcesToMoveDown = realm.objects(RealmFavoritedResource.self).where( { $0.position <= originalPosition && $0.position >= newPosition})
+                for resource in resourcesToMoveDown {
+                    resource.position += 1
+                }
+                
+                resourcesToUpdate += resourcesToMoveDown
+            }
+            
+            if let resourceToMove = realm.object(ofType: RealmFavoritedResource.self, forPrimaryKey: id) {
+                resourceToMove.position = newPosition
+                resourcesToUpdate.append(resourceToMove)
+            }
+            
+            return resourcesToUpdate
+            
+        } mapInBackgroundClosure: { (objects: [RealmFavoritedResource]) in
+            return objects.map({
+                FavoritedResourceDataModel(realmFavoritedResource: $0)
+            })
+        }
+        .eraseToAnyPublisher()
+    }
+    
+    // MARK: - Private
+    
+    private func validatePositionsForFavoritedResources(ascendingOrder: Bool) -> AnyPublisher<[FavoritedResourceDataModel], Never> {
+        let favoritesSortedByPosition = self.realmDatabase.openRealm()
+            .objects(RealmFavoritedResource.self)
+            .sorted(byKeyPath: #keyPath(RealmFavoritedResource.position), ascending: ascendingOrder)
+        
+        lazy var favoritesDataModels: [FavoritedResourceDataModel] = favoritesSortedByPosition.map({
```

Comment:
Ah didn't realize this was marked ```lazy```, at first I was thinking you created some extra overhead, but with ```lazy``` this wouldn't get invoked.  Was about to comment, but checked and looks good.